### DISCUSSION TITLE
searx - apache2 - http 2.0 support

### DISCUSSION THREAD

**Discussion by asteriosos** [searx/searx#1539] [Installation issue: Official wiki]
hi,

there is no http 2.0 support with apache2 in searx or is it a generally problem ?

if i enable it in apache2, searx dosen't work anymore, only in http 1.1.
i get the website sourcecode displayed with an header: says http 2.0 200 ok.

i installed it with the documentation running apache2 (debian 9) and uwsgi.

thank you.

**Comment by dalf** [comment]
The HTTP2 support is not related to searx.
The configuration should look like this one ( https://asciimoo.github.io/searx/dev/install/installation.html  and https://httpd.apache.org/docs/2.4/howto/http2.html)
```
<VirtualHost ...>
    ServerName test.example.org
    Protocols h2 http/1.1
    ...
    <Location />
      Options FollowSymLinks Indexes
      SetHandler uwsgi-handler
      uWSGISocket /run/uwsgi/app/searx/socket
    </Location>
    ...
</VirtualHost>
```

what is the content of the uwsgi log file, errors.log (the one from Apache HTTP) ?
if there is nothing revelant, try to set ```debug``` to ```True``` in searx/settings.yml

**Comment by asteriosos** [comment]
apache2 error.log:
[Fri Mar 29 15:03:19.706880 2019] [:error] [pid 4739:tid 140450502452992] [client 109.91.36.17:23660] uwsgi: max number of uwsgi variables reached. consider increasing it with uWSGImaxVars directive

i dont get this with h2 disabled...

i set uWSGImaxVars up to 256 the error is  gone, but still cannot use searx.

searx debug log:
[searx.log](https://github.com/asciimoo/searx/files/3022984/searx.log)

latest searx, debian 9.8, apache 2.4.25, mod_http2 v1.10.16

**Comment by asteriosos** [comment]
At the moment i disabled http2 support on apache 2.4.25 there are some security issues reported with http2 enabled.

**Comment by asteriosos** [comment]
I got an update on h2 with debian 9.8 yesterday. I enabled h2 again and i still get this issue.
